### DISCUSSION TITLE
count vs findOne ? Which one is efficient

### DISCUSSION THREAD

**Discussion by uzair004** [sequelize/sequelize#14340] [Help & Questions]
Most of the time we want to check if something exist in database or not.

There are 3 main approaches in SQL in general

1. find single record (findOne in sequelize)
2. count records if greater then zero (count() method in sequelize.
3. Exists SQL clause (Not available in sequelize)

Among first two which one should be used? 
Don't you think there should be `exists()` method in sequelize?

**Comment by ephys** [comment]
`EXISTS` is a subquery operator. Sequelize has poor support for subqueries (that's changing) so you'd have to write the query in raw SQL:

```
User.findAll({
  where: literal('EXISTS (select id from users)')
})
```

The fastest solution is probably this:
```js
User.findOne({
  attributes: ['id'], // reduce how much data is loaded
  raw: true, // don't build a model instance, return the raw data
});
```

**Comment by ephys** [comment]
Found this issue that might interest you https://github.com/sequelize/sequelize/issues/10187

**Comment by alacret** [comment]
@ephys @uzair004 

After doing some tests directly with MySQL, I found that a query with a LIMIT 1 is up to 10 times faster than a count on the table without a WHERE clause.

**Reply by uzair004** [reply]
Can you explain that? how do you get count using that?

**Reply by alacret** [reply]
@uzair004 

In the specific of knowing if a table has records or not:

Instead of doing

```typescript
  const count = await Model.count();
 console.log( count === 0)
```

It is faster to do:

```typescript
  const oneRecord = await Model.findAll({    limit: 1  });
 console.log( oneRecord === null)
```
